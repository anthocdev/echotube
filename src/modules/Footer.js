import React from "react";
import Link from "@material-ui/core/Link";
import { Typography } from "@material-ui/core";
/* Simple Footer with copyright notice */
function CopyrightFooter() {
  return (
    <Typography variant="body2" style={{ color: "#ffffff" }} align="center">
      {"Copyright Â© "}
      <Link color="inherit" href="http://cerneckis.com/">
        Antanas Cerneckis <br />
      </Link>
      {new Date().getFullYear()}
      {"."}
    </Typography>
  );
}

export default function Footer() {
  return (
    <div
      style={{
        position: "absolute",
        left: 0,
        bottom: 0,
        right: 0,
        backgroundColor: "rgba(153, 50, 204, 0.6)",
      }}
    >
      <CopyrightFooter />
    </div>
  );
}
