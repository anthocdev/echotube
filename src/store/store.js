import { applyMiddleware, createStore, compose } from "redux";
import { createLogger } from "redux-logger";
import { default as thunk } from "redux-thunk";
import { createPromise } from "redux-promise-middleware";
import axios from "axios";

import reducer from "../reducers";
//Pulling JWT Token from local storage
const jwttoken = localStorage.getItem("JWT_TOKEN");
//Setting JWT Token as common header for axios
axios.defaults.headers.common["Authorization"] = jwttoken;
//Create store with middleware included
const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose; //Adding redux dev tool compatibility
const middleware = applyMiddleware(createPromise(), thunk, createLogger()); //Adding thunk and console logging for testing as middleware

export default createStore(
  reducer,
  {
    //Initializing auth state if JWT token is available
    auth: {
      jwtToken: jwttoken,
      isAuth: jwttoken ? true : false
    }
  },
  composeEnhancers(middleware)
);
