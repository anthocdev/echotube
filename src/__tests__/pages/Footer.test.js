import React from "react";
import Footer from "../../modules/Footer";
import { Provider } from "react-redux";
import store from "../../store/store";
import { create, act } from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";

describe("Footer Module Tests", () => {
  let root;

  it("Renders Without Crashing", async () => {
    act(() => {
      root = create(
        <Provider store={store}>
          <Footer />
        </Provider>
      );
    });
  });

  test("Renders Matches Snapshot", async () => {
    act(() => {
      root = create(
        <BrowserRouter>
          <Provider store={store}>
            <Footer />
          </Provider>
        </BrowserRouter>
      );
    });
    expect(root).toMatchSnapshot();
    console.log(root.toJSON());
  });
});
