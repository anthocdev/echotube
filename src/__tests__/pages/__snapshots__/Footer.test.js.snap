// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Module Tests Renders Matches Snapshot 1`] = `
<div
  style={
    Object {
      "backgroundColor": "rgba(153, 50, 204, 0.6)",
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
    }
  }
>
  <p
    className="MuiTypography-root MuiTypography-body2 MuiTypography-alignCenter"
    style={
      Object {
        "color": "#ffffff",
      }
    }
  >
    Copyright Â© 
    <a
      className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
      href="http://cerneckis.com/"
      onBlur={[Function]}
      onFocus={[Function]}
    >
      Antanas Cerneckis 
      <br />
    </a>
    2020
    .
  </p>
</div>
`;
